docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc

```diff
@@ -0,0 +1,213 @@
+As soon as the container starts, {agent} checks it for hints and launches the proper configuration
+for the container. Hints tell {agent} how to monitor the container by using the proper integration.
+This is the full list of supported hints:
+
+[discrete]
+== Required hints:
+
+[float]
+=== `co.elastic.hints/package`
+
+The package to use for monitoring.
+
+[float]
+=== `co.elastic.hints/host`
+
+The host to use for metrics retrieval.
+
+[discrete]
+== Optional hints available:
+
+[float]
+=== `co.elastic.hints/data_stream`
+
+The list of data streams to enable. If not specified, the integration's default data streams are used.
+To find the defaults, refer to the {integrations-docs}[Elastic integrations documentation].
+
+If data streams are specified, additional hints can be defined per data stream. 
+
+For example, `co.elastic.hints/info.period: 5m` if the data stream specified is `info` for the {metricbeat-ref}/metricbeat-module-redis.html[Redis module]. 
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: redis
+  annotations:
+    co.elastic.hints/package: redis
+    co.elastic.hints/data_streams: info
+    co.elastic.hints/info.period: 5m
+----
+
+(`co.elastic.hints/period: 10m`) in its configuration.
+
+[float]
+=== `co.elastic.hints/metrics_path`
+
+The path to retrieve the metrics from.
+
+[float]
+=== `co.elastic.hints/period`
+
+The time interval for metrics retrieval, for example, 10s.
+
+[float]
+=== `co.elastic.hints/timeout`
+
+Metrics retrieval timeout, for example, 3s.
+
+[float]
+=== `co.elastic.hints/username`
+
+The username to use for authentication.
+
+[float]
+=== `co.elastic.hints/password`
+
+The password to use for authentication. It is recommended to retrieve this sensitive information from an ENV variable
+and avoid placing passwords in plain text.
+
+[float]
+=== `co.elastic.hints/stream`
+
+The stream to use for logs collection, for example, stdout/stderr.
+
+If the specified package has no logs support, a generic container's logs input will be used as a fallback.
+
+[discrete]
+== Available packages that support hints autodiscovery
+
+The available packages that are supported through hints can be found
+https://github.com/elastic/elastic-agent/tree/main/deploy/kubernetes/elastic-agent-standalone/templates.d[here].
+
+[discrete]
+== Configure hints autodiscovery
+
+To enable hints, you must add `hints.enabled: true` to the provider's configuration:
+
+[source,yaml]
+----
+providers:
+  kubernetes:
+    hints.enabled: true
+----
+
+Then ensure that the proper volumes and volumeMounts are specified by uncommenting the appropriate sections in the {agent} manifest:
+
+[source,yaml]
+----
+volumeMounts:
+- name: external-inputs
+  mountPath: /etc/elastic-agent/inputs.d
+...
+volumes:
+- name: external-inputs
+  emptyDir: {}
+...
+----
+
+An init container is also required to download the hints templates.
+The init container is already defined, so uncomment the respective section:
```

Comment:
I am a little hesitant to do this, because if the manifest changes we don't have any guarantees that the line with redis will remain the same and pointing to a different one could be confusing. What do you think?

Reply:
I thought about link to the manifest itself - https://github.com/elastic/elastic-agent/blob/main/deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml, not to the specific line(s) of init container - https://github.com/elastic/elastic-agent/blob/main/deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml#L657-L668
but I think that this manifest was mentioned already in documentation, so it might be indeed redundant. Let's keep it as is